src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -57,7 +57,12 @@ def fence_old_ranks(self,
     def config(self, spec: NFSServiceSpec, daemon_id: str) -> None:  # type: ignore
         assert self.TYPE == spec.service_type
         assert spec.pool
-        self.mgr._check_pool_exists(spec.pool, spec.service_name())
+        pool_list = [p['pool_name'] for p in self.mgr.get_osdmap().dump().get('pools', [])]
+
+        if spec.pool not in pool_list:
+            self.mgr.check_mon_command({'prefix': 'osd pool create', 'pool': spec.pool})
+            self.mgr.check_mon_command({'prefix': 'osd pool application enable',
+                                        'pool': spec.pool, 'app': 'nfs'})
```

Comment:
Move this to orchestrator code instead of being specific to cephadm ?

Reply:
If we plan to use `nfs-ganesha` as default pool. Then we can remove redundant code from nfs module too
https://github.com/ceph/ceph/blob/master/src/pybind/mgr/nfs/cluster.py#L68-L73